server/src/main/java/org/elasticsearch/common/compress/Compressor.java

```diff
@@ -32,11 +32,19 @@
 
     int headerLength();
 
+    /**
+     * Creates a new stream input that decompresses the contents read from teh provided stream input.
+     * Closing the returned {@link StreamInput} will close the provided stream input.
+     * Note: The returned stream may only be used on the thread that created it as it might use thread-local resources and must be safely
+     * closed after use
+     */
     StreamInput streamInput(StreamInput in) throws IOException;
```

Comment:
Maybe name this method something like `unsafeStreamInput` or `threadLocalStreamInput`?